packages/aws-cdk-lib/aws-s3/lib/bucket.ts

```diff
@@ -2127,12 +2129,18 @@ export class Bucket extends BucketBase {
       throw new Error(`encryptionKey is specified, so 'encryption' must be set to KMS or DSSE (value: ${encryptionType})`);
     }
 
-    // if bucketKeyEnabled is set, encryption must be set to KMS or DSSE.
+    // if bucketKeyEnabled is set, encryption must be set to KMS, DSSE or S3.
     if (
       props.bucketKeyEnabled &&
-      ![BucketEncryption.KMS, BucketEncryption.KMS_MANAGED, BucketEncryption.DSSE, BucketEncryption.DSSE_MANAGED].includes(encryptionType)
+      ![
+        BucketEncryption.KMS,
+        BucketEncryption.KMS_MANAGED,
+        BucketEncryption.DSSE,
+        BucketEncryption.DSSE_MANAGED,
+        BucketEncryption.S3_MANAGED,
+      ].includes(encryptionType)
```

Comment:
I see in the `BucketEncryption` enum that the error is now only thrown when the encryptionType is `UNENCRYPTED`. Should this check be changed to throw the error if `encryptionType == BucketEncryption.UNENCRYPTED`?

Reply:
Agree!